test/integration/DCAHubCompanion/multi-call.spec.ts

```diff
@@ -87,15 +85,210 @@ contract('Multicall', () => {
+      });
+      swapAndDepositTest({
+        from: 'USDC',
+        swap: ({ amountIn }) => swapIn0x({ from: USDC, to: WETH, amountIn }),
+      });
+      function swapAndDepositTest({ from, swap }: { from: 'USDC' | 'ETH'; swap: Swap }) {
+        const AMOUNT_IN = USDC_1000;
+        when(`swapping ${from} to WETH and depositing`, () => {
+          let minExpected: BigNumber;
+          given(async () => {
+            const takeData = await takeFromCallerDataIfUSDC({ from, amount: USDC_1000 });
+            const { swapExecutionData, expectedAmountOut } = await runSwapData({ tokenIn: await getAddress(from), amountIn: AMOUNT_IN, swap });
+            const depositData = await depositAllInCompanionData({ from: WETH, to: WBTC });
+            await DCAHubCompanion.multicall(filterMulticalls([takeData, swapExecutionData, depositData]), {
+              value: from === 'ETH' ? AMOUNT_IN : 0,
+            });
+            minExpected = expectedAmountOut;
+          });
+          then('hub has expected amount of WETH balance', async () => {
+            const balance = await WETH.balanceOf(DCAHub.address);
+            expect(balance.gte(minExpected)).to.be.true;
+          });
+          then('new position is created', async () => {
+            await expectPositionToHaveBeenCreated({ positionId: 1, from: WETH, to: WBTC, minRemaining: minExpected });
+          });
+          thenCompanionRemainsWithoutAnyBalance();
+        });
+      }
+    });
 
-    describe('swap and increase', () => {});
+    describe('swap and increase', () => {
+      swapAndIncreaseTest({
+        from: 'ETH',
+        swap: ({ amountIn }) => transformETHToWETH(amountIn),
+      });
+      swapAndIncreaseTest({
+        from: 'USDC',
+        swap: ({ amountIn }) => swapIn0x({ from: USDC, to: WETH, amountIn }),
+      });
+      function swapAndIncreaseTest({ from, swap }: { from: 'USDC' | 'ETH'; swap: Swap }) {
+        const AMOUNT_IN = USDC_1000;
+        when(`swapping ${from} to WETH and increasing a position`, () => {
+          let positionId: BigNumber, minExpectedBalance: BigNumber;
+          given(async () => {
+            const { positionId: createdPositionId, unswappedBalance } = await depositAndSwap({ from: WETH, to: USDC, amount: ETH_1 });
+            const takeData = await takeFromCallerDataIfUSDC({ from, amount: USDC_1000 });
+            const { swapExecutionData, expectedAmountOut } = await runSwapData({ tokenIn: await getAddress(from), amountIn: AMOUNT_IN, swap });
+            const permissionData = await givePermissionToCompanionData({
+              signer: positionOwner,
+              positionId: createdPositionId,
+              permissions: [Permission.INCREASE],
+            });
+            const increaseData = await increaseAllInCompanionData({ positionId: createdPositionId });
+            await DCAHubCompanion.multicall(filterMulticalls([takeData, swapExecutionData, permissionData, increaseData]), {
```

Comment:
I don't get what `filterMulticalls` is intended for.

Reply:
`takeData` might be `undefined`, since we don't need to take anything in the case of `from == ETH`. So `filterMulticalls` just removed said `undefined` from the array